drivers/ad5770r/ad5770r.c

```diff
@@ -0,0 +1,650 @@
+				 bool *mask_channel_sel)
+{
+	int32_t ret;
+
+	ret = ad5770r_spi_reg_write(dev,
+				    AD5770R_CH_SELECT,
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel[0], AD5770R_CH0) |
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel[1], AD5770R_CH1) |
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel[2], AD5770R_CH2) |
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel[3], AD5770R_CH3) |
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel[4], AD5770R_CH4) |
+				    AD5770R_CH_SELECT_SEL_CH(mask_channel_sel[5], AD5770R_CH5));
+
+	if (ret)
+		return -1;
+
+	dev->mask_channel_sel = mask_channel_sel;
+
+	return ret;
+}
+
+/**
+ * Set software LDAC.
+ * @param dev - The device structure.
+ * @param sw_ldac - The array contains channel selection options.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_set_sw_ldac(struct ad5770r_dev *dev,
+			    bool *sw_ldac)
+{
+	int32_t ret;
+
+	ret = ad5770r_spi_reg_write(dev,
+				    AD5770R_SW_LDAC,
+				    AD5770R_SW_LDAC_CH(sw_ldac[0], AD5770R_CH0) |
+				    AD5770R_SW_LDAC_CH(sw_ldac[1], AD5770R_CH1) |
+				    AD5770R_SW_LDAC_CH(sw_ldac[2], AD5770R_CH2) |
+				    AD5770R_SW_LDAC_CH(sw_ldac[3], AD5770R_CH3) |
+				    AD5770R_SW_LDAC_CH(sw_ldac[4], AD5770R_CH4) |
+				    AD5770R_SW_LDAC_CH(sw_ldac[5], AD5770R_CH5));
+
+	if (ret)
+		return -1;
+
+	dev->sw_ldac = sw_ldac;
+
+	return ret;
+}
+
+/**
+ * Set the enabled channels.
+ * @param dev - The device structure.
+ * @param channel_enable - The array contains channels enabled.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_set_channel_en(struct ad5770r_dev *dev,
+			       bool *channel_enable)
+{
+	int32_t ret;
+
+	ret = ad5770r_spi_reg_write(dev,
+				    AD5770R_CH_ENABLE,
+				    AD5770R_CH_ENABLE_SET(channel_enable[0], AD5770R_CH0) |
+				    AD5770R_CH_ENABLE_SET(channel_enable[1], AD5770R_CH1) |
+				    AD5770R_CH_ENABLE_SET(channel_enable[2], AD5770R_CH2) |
+				    AD5770R_CH_ENABLE_SET(channel_enable[3], AD5770R_CH3) |
+				    AD5770R_CH_ENABLE_SET(channel_enable[4], AD5770R_CH4) |
+				    AD5770R_CH_ENABLE_SET(channel_enable[5], AD5770R_CH5));
+
+	if (ret)
+		return -1;
+
+	dev->channel_enable = channel_enable;
+
+	return ret;
+}
+
+/**
+ * Get status value.
+ * @param dev - The device structure.
+ * @param mon_setup - The structure that contains the monitor setup
+ *		values.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_get_status(struct ad5770r_dev *dev,
+			   uint8_t *status)
+{
+	return ad5770r_spi_reg_read(dev,
+				    AD5770R_STATUS,
+				    status);
+};
+
+/**
+ * Set the channel monitor configuration.
+ * @param dev - The device structure.
+ * @param mon_setup - The structure that contains the monitor setup
+ *		values.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_set_monitor_setup(struct ad5770r_dev *dev,
+				  struct monitor_setup mon_setup)
```

Comment:
pass this via pointer; this also allocs `sizeof(struct monitor_setup)` on the call-stack